packages/core/src/components/Dropdown/Dropdown.tsx

```diff
@@ -18,16 +17,284 @@ import {
+  /**
+   * ID for the select container
+   */
+  id?: DROPDOWN_MENU.ID;
+  /**
+   * focusAuto when component mount
+   */
+  autoFocus?: boolean;
+  /**
+   * If set to true, the dropdown will be in multi-select mode.
+   * When in multi-select mode, the selected value will be an array,
+   * and it will be displayed as our [`<Chips>`](/?path=/docs/components-chips--sandbox) component.
+   */
+  multi?: boolean;
+  /**
+   * If set to true together with `multi`, it will make the dropdown expand to multiple lines when new values are selected.
+   */
+  multiline?: boolean;
+  /**
+  Pass closeMenuOnSelect to close the multi choose any time an options is chosen.
+  */
+  closeMenuOnSelect?: boolean;
+  // Won't be needed once we upgrade to react-select ^5.5 https://github.com/JedWatson/react-select/issues/4088#issuecomment-1276835389
+  /**
+   * If menu should be closed on scroll - helpful for some tricky use cases
+   * @default false, but true when insideOverflowContainer or insideOverflowWithTransformContainer are true
+   */
+  closeMenuOnScroll?: ((event: React.FocusEvent) => boolean) | boolean;
+  /**
+   * callback to be called when `multiline` is `true` and the option is removed
+   */
+  onOptionRemove?: (...args: unknown[]) => unknown;
+  /**
+  Pass Ref for reference of the actual dropdown component
+  */
+  ref?: React.ForwardedRef<HTMLDivElement>;
+  /**
+  The options set by default will be set as mandatory and the user will not be able to cancel their selection
+  */
+  withMandatoryDefaultOptions?: boolean;
+  /**
+   * Override the built-in logic to detect whether an option is selected.
+   */
+  isOptionSelected?: (option: DropdownDefaultValue, selectValue: DropdownDefaultValue["value"]) => boolean;
+  /**
+   * For display the drop down menu in overflow hidden/scroll container.
+   */
+  insideOverflowContainer?: boolean;
+  /**
+   * For display the drop down menu in overflow hidden/scroll container which contains transform css function usage.
+   */
+  insideOverflowWithTransformContainer?: boolean;
+  /**
+   * When content is passed, the dropdown will include a tooltip on the dropdown's value.
+   */
+  tooltipContent?: string;
+  /**
+   * Display the drop down with loading state.
+   */
+  isLoading?: boolean;
+  /**
+   * Overrides the built-in logic of loading message design
+   */
+  loadingMessage?: (obj: { inputValue: string }) => React.ReactNode;
+  /**
+   * aria-label attribute for dropdown
+   */
+  ariaLabel?: string;
+  /**
+   * Overrides the built-in logic of tab selecting value (default: true)
+   */
+  tabSelectsValue?: boolean;
+  /**
+   * Overrides the build-in search filter logic - https://react-select.com/advanced#custom-filter-logic
+   * createFilter function is available at Dropdown.createFilter
+   */
+  filterOption?: (option: unknown, inputValue: string) => boolean;
 
-const Dropdown = forwardRef(
+  withReadOnlyStyle?: boolean;
+  OptionRenderer?: (option: IComboboxOption) => JSX.Element;
```

Comment:
Why ComboboxOption

Reply:
I took it from Combobox.tsx,
changing to DropdownOption